Public/Src/Tools/SandboxedProcessExecutor/Executor.cs

```diff
@@ -0,0 +1,322 @@
+            }
+
+            Environment.Exit((int)ExitCode.InternalError);
+        }
+
+        private void TelemetryStartup()
+        {
+            if (!Debugger.IsAttached && m_configuration.EnableTelemetry)
+            {
+                AriaV2StaticState.Enable(global::BuildXL.Tracing.AriaTenantToken.Key);
+                TrackingEventListener.RegisterEventSource(ETWLogger.Log);
+                m_telemetryStopwatch.Start();
+            }
+        }
+
+        private void TelemetryShutdown()
+        {
+            if (AriaV2StaticState.IsEnabled && m_configuration.EnableTelemetry)
+            {
+                m_telemetryStopwatch.Stop();
+                Logger.Log.SandboxedProcessExecutorInvoked(m_loggingContext, m_telemetryStopwatch.ElapsedMilliseconds, Environment.CommandLine);
+                AriaV2StaticState.TryShutDown(TimeSpan.FromSeconds(10), out var telemetryShutdownException);
+            }
+        }
+
+        internal ExitCode RunInternal()
+        {
+            if (!TryReadSandboxedProcessInfo(out SandboxedProcessInfo sandboxedProcessInfo))
+            {
+                return ExitCode.FailedReadInput;
+            }
+
+            if (!TryPrepareSandboxedProcess(sandboxedProcessInfo))
+            {
+                return ExitCode.FailedSandboxPreparation;
+            }
+
+            (ExitCode exitCode, SandboxedProcessResult result) executeResult = ExecuteAsync(sandboxedProcessInfo).GetAwaiter().GetResult();
+
+            if (executeResult.result != null)
+            {
+                if (!TryWriteSandboxedProcessResult(executeResult.result))
+                {
+                    return ExitCode.FailedWriteOutput;
+                }
+            }
+
+            return executeResult.exitCode;
+        }
+
+        private bool TryReadSandboxedProcessInfo(out SandboxedProcessInfo sandboxedProcessInfo)
+        {
+            sandboxedProcessInfo = null;
+
+            try
+            {
+                using (FileStream stream = File.OpenRead(Path.GetFullPath(m_configuration.SandboxedProcessInfoInputFile)))
+                {
+                    // TODO: Custom DetoursEventListener?
+                    sandboxedProcessInfo = SandboxedProcessInfo.Deserialize(stream, m_loggingContext, detoursEventListener: null);
+                }
+            }
+            catch (IOException ioException)
+            {
+                ConsoleError(ioException.ToString());
+                return false;
+            }
+
+            return true;
+        }
+
+        private bool TryWriteSandboxedProcessResult(SandboxedProcessResult result)
+        {
+            Contract.Requires(result != null);
+
+            try
+            {
+                using (FileStream stream = File.OpenWrite(Path.GetFullPath(m_configuration.SandboxedProcessResultOutputFile)))
+                {
+                    result.Serialize(stream);
+                }
+            }
+            catch (IOException ioException)

```

Comment:
>IOException [](start = 19, length = 11)

Wherever you catch IOException in this file you must also catch UnauthorizedAccessException